incubator/mongodb/templates/mongodb-petset.yaml

```diff
@@ -0,0 +1,136 @@
+        pod.alpha.kubernetes.io/initialized: "true"
+        pod.alpha.kubernetes.io/init-containers: '[
+            {
+                "name": "install",
+                "image": "gcr.io/google_containers/mongodb-install:0.1",
+                "args": ["--work-dir=/work-dir"],
+                "volumeMounts": [
+                    {
+                        "name": "workdir",
+                        "mountPath": "/work-dir"
+                    },
+                    {
+                        "name": "config",
+                        "mountPath": "/config"
+                    }
+                ]
+            },
+            {
+                "name": "bootstrap",
+                "image": "{{.Values.Image}}:{{.Values.ImageTag}}",
+                "command": ["/work-dir/peer-finder"],
+                "args": ["-on-start=\"/work-dir/on-start.sh\"", "-service={{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"],
+                "env": [
+                  {
+                      "name": "POD_NAMESPACE",
+                      "valueFrom": {
+                          "fieldRef": {
+                              "apiVersion": "v1",
+                              "fieldPath": "metadata.namespace"
+                          }
+                      }
+                   }
+                ],
+                "volumeMounts": [
+                    {
+                        "name": "workdir",
+                        "mountPath": "/work-dir"
+                    },
+                    {
+                        "name": "config",
+                        "mountPath": "/config"
+                    },
+                    {
+                        "name": "datadir",
+                        "mountPath": "/data/db"
+                    }
+                ]
+            }
+        ]'
+    spec:
+      containers:
+      - name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+        image: "{{.Values.Image}}:{{.Values.ImageTag}}"
+        imagePullPolicy: "{{.Values.ImagePullPolicy}}"
+        ports:
+        - containerPort: {{.Values.PeerPort}}
+          name: peer
+        resources:
+          requests:
+            cpu: "{{.Values.Cpu}}"
+            memory: "{{.Values.Memory}}"
+        command:
+        - /usr/bin/mongod
+        - --config=/config/mongod.conf
+        readinessProbe:
+          exec:
+            command:
+            - sh
+            - -c
+            - "/usr/bin/mongo --eval 'printjson(db.serverStatus())'"
+          initialDelaySeconds: 5
+          timeoutSeconds: 5
+        volumeMounts:
+        - name: datadir
+          mountPath: /data/db
+        - name: config
+          mountPath: /config
+      volumes:
+      - name: config
+        emptyDir: {}
+      - name: workdir
+        emptyDir: {}
+  volumeClaimTemplates:
+  - metadata:
+      name: datadir
+      annotations:
+        volume.alpha.kubernetes.io/storage-class: anything
```

Comment:
we want to allow for the setting of storage-class with a parameter.  Yay ec2 IOPS :)